### DISCUSSION TITLE
Remember the current file's caret position when quitting/starting Obsidian

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#175] [Show and tell]
There's already a [community plugin for Obsidian to remember the caret position](https://github.com/dy-sh/obsidian-remember-cursor-position), but - just for fun - I made two shell commands to accomplish a simpler "poor man's version" version of it:

https://publish.obsidian.md/shellcommands/Example+shell+commands/Remember+caret+position+when+reopening+Obsidian

**Comment by 668168** [comment]
great, how to open the position on next start obsidian ?

**Reply by Taitava** [reply]
Thanks for your question 🙂. What operating system are you using? 🙂 

The second shell command that is mentioned in the documentation's example should open the current file at the correct position when Obsidian starts:
For Windows: ``Write-Host {{file_path:relative}}`: -NoNewLine; type caret_position.txt``
For Linux and macOS: `echo -n {{file_path:relative}}\:; cat caret_position.txt`

(If you are using Windows, you need to define your shell to be _PowerShell 5_ or _PowerShell Core_. Here's a guide: https://publish.obsidian.md/shellcommands/Operating+systems+and+shells/Shells#Choosing+a+shell+on+a+per+shell+command+basis)

Then you need to enable an event _After Obsidian starts_ for the shell command, and set `stdout` output channel to *[Open a file](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Open+a+file)*. Here's how you can select an output channel: https://publish.obsidian.md/shellcommands/Output+handling/Output+channels#How+to+select+output+channels+for+a+shell+command

Please let me know if you still have any questions! 🙂

**Comment by 668168** [comment]
yeah! 
the echo **-n** is so smart method.  i has spend a long time to understand it!

thank , Taitava, you are so nice man.

**Reply by Taitava** [reply]
Thank you, nice that I can be of help! 😎
